### PR TITLE
Add specific errors for client socket access (closes 595)

### PR DESCRIPTION
**Pull request by leeclemens**
Add os.access() calls and specific error messages regarding socket access

### CODE REVIEW

File: bin/fail2ban-client [hunk 1]
```diff
@@ -144,32 +144,48 @@ class Fail2banClient:
...
-		for c in cmd:
-			beautifier.setInputCmd(c)
-			try:
-				client = CSocket(self.__conf["socket"])
-				ret = client.send(c)
-				if ret[0] == 0:
-					logSys.debug("OK : " + `ret[1]`)
+		streamRet = False
+		if self.__hasSocketAccess(showRet):
+			beautifier = Beautifier()
```
Comment (yarikoptic) [inline bin/fail2ban-client]: imho overindentation makes code more difficult to parse... why not
```python
if not self.__hasSocketAccess(showRet):
    return False
# continue the main logic
```
?

**Comment by sebres** [comment]
Because `fail2ban-client` can be started multiple times, I believe, the checking `__hasSocketAccess` in error case only, would be much better solution.
I mean for example in `except` scope. Please, makes f2b not slower, than it already is.

File: bin/fail2ban-client (hunk 1 above)
Comment (sebres) [inline bin/fail2ban-client]: I think, it should be at all not executed, before any error occured... (keyword: performance).

**Comment by leeclemens** [comment]
@sebres I see your point, and I initially thought about adding all of that logic into the `except socket.error` block. However, I tend to shy away from any more logic (potentially creating an error/exception) in an `except` block than necessary.

File: bin/fail2ban-client (hunk 1 above)
Comment (leeclemens) [inline bin/fail2ban-client]: You're right, for the wrong reason (imho) :)   I made my change thinking __processCmd "processed a command"...negating the loop of potentially multiple commands...any of which could fail or succeed independently and during which the socket could vanish

**Comment by leeclemens** [comment]
Okay, new logic only occurs if socket.error was raised and showRet==True. Wrapped in its own try block, so it should be safe.

**Comment by sebres** [comment]
:+1:

**Pull request merged**